File: advent of code/2021/python/day1.py
# Advent of Code 2021 - Day 1: Sonar Sweep
# 1 December 2021
# 4:25 PM - 5:27 PM, on off at work

# reading input
f = open('inputs/day1.txt', 'r')
lines = f.readlines()

# part 1
counter = index = 0

for line in lines:

    if(index != 0):
        current = line.rstrip()
        previous = (lines.__getitem__(index-1)).rstrip()    

        if(int(current) > int(previous)): 
            counter += 1
    index += 1

print(counter) # part 1 answer

# part 2
sums = []

for i in range(lines.__len__()-2):
    sums.append(int((lines.__getitem__(i)).rstrip()) + 
                int((lines.__getitem__(i+1)).rstrip()) + 
                int((lines.__getitem__(i+2)).rstrip()))

counter = index = 0

for sum in sums:

    if(index != 0):
        current = sum
        previous = sums.__getitem__(index-1)

        if(int(current) > int(previous)): 
            counter += 1
    index += 1

print(counter) # part 2 answer

File: advent of code/2021/java/day1.java
/**
 * Advent of Code 2021 - Day 1: Sonar Sweep
 * 30 November 2021
 * 11:01 PM - 11:20 PM
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class day1 {
    public static void main(String[] args) throws FileNotFoundException {
        part1(); // 13 minutes
        part2(); // 6 minutes
    }

    public static void part1() throws FileNotFoundException {
        File input = new File("inputs/day1.txt");

        Scanner sc = new Scanner(input);
        ArrayList<Integer> measurements = new ArrayList<>();

        while(sc.hasNextLine()) {
            measurements.add(sc.nextInt());
        }

        //
        //System.out.println(measurements.size());

        int counter = 0;
        int index = 0;

        for(int measurement : measurements) {
            if(index == 0) { // onlny happens at the beginning
                index++;
            } else {
                if(measurement > measurements.get(index-1)) { // w/o index-1, comparing measurement to itself
                    counter++;
                }
                index++;
            }
        }

        sc.close();

        System.out.println(counter);
    }

    public static void part2() throws FileNotFoundException {
        File input = new File("inputs/day1.txt");

        Scanner sc = new Scanner(input);
        ArrayList<Integer> measurements = new ArrayList<>();

        while(sc.hasNextLine()) {
            measurements.add(sc.nextInt());
        }

        ArrayList<Integer> sums = new ArrayList<>();

        // getting all the sums
        for(int i = 0; i < measurements.size() - 2; i++) {
            sums.add(measurements.get(i) + measurements.get(i+1) + measurements.get(i+2));
        }

        //System.out.println(sums.size());

        int counter = 0;
        int index = 0;

        for(int sum : sums) {
            if(index == 0) { // onlny happens at the beginning
                index++;
            } else {
                if(sum > sums.get(index-1)) { // w/o index-1, comparing measurement to itself
                    counter++;
                }
                index++;
            }
        }

        sc.close();

        System.out.println(counter);
    }
}